### PR TITLE
Make it compatible with Webpack 3.0 and update dependencies

### PR DESCRIPTION
**Pull request by vlucendo**
When used with the latest Webpack it shows:

> DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56 parseQuery() will be replaced with getOptions() in the next major version of loader-utils.

I've updated dependencies, documentation and made some changes to make it work properly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +85,7 @@ As a result, `renderedHtml` becomes a string `<h1><a href="http://example.com">E
 
 
 ## Release History
+* 0.3.1 - Make it compatible with Webpack 3.0 and update dependencies
```
Comment (okonet) [inline README.md]: This will be a breaking change and thus needs to be a major version.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ejs-loader",
-  "version": "0.3.0",
+  "version": "0.3.1",
```
Comment (okonet) [inline package.json]: Please remove this from PR since I’m using a script to automate this.

File: README.md [hunk 3]
```diff
@@ -10,35 +10,25 @@ If you are looking for the loader which uses [EJS templating engine](https://git
 
 ## Usage
 
-[Documentation: Using loaders](http://webpack.github.io/docs/using-loaders.html)
+[Documentation: Using loaders](https://webpack.js.org/concepts/loaders/)
 
 ``` javascript
-var template = require("ejs!./file.ejs");
+var template = require("./file.ejs");
```
Comment (okonet) [inline README.md]: Why did you remove the loader part? This example won’t work out of the box.

File: README.md [hunk 4]
```diff
@@ -10,35 +10,25 @@ If you are looking for the loader which uses [EJS templating engine](https://git
...
 // => returns the template function compiled with undesrcore (lodash) templating engine.
 
 // And then use it somewhere in your code
 template(data) // Pass object with data
 ```
 
-You also should provide a global `_` variable with the lodash/underscore runtime. You can do it with the following webpack plugin: https://github.com/webpack/docs/wiki/list-of-plugins#provideplugin
```
Comment (okonet) [inline README.md]: Not sure why this was removed either

File: README.md (hunk 1 above)
Comment (vlucendo) [reply README.md]: True, will change it.

File: package.json (hunk 2 above)
Comment (vlucendo) [reply package.json]: Ok

File: README.md (hunk 4 above)
Comment (vlucendo) [reply README.md]: It didn't work for me because it produced an error as I pasted the readme example that uses underscore instead of lodash in my config.
Also, since my project didn't require any escaping it didn't show any errors.
Fixed.

File: README.md (hunk 3 above)
Comment (okonet) [reply README.md]: Yes, it's not required if there is a config but for the sake of Readme I don't want to complicate things. Please revert.

File: README.md [hunk 5]
```diff
@@ -6,43 +6,55 @@ If you are looking for the loader which uses [EJS templating engine](https://git
...
     new webpack.ProvidePlugin({
-        _: "underscore"
+        _: "lodash"
     })
 ]
 ```
 
 ### Options
-[Underscore](http://underscorejs.org/#template)/[Lodash](https://lodash.com/docs#template) options can be passed in using the querystring or adding an ```esjLoader``` options block to your configuration.
+[Underscore](http://underscorejs.org/#template)/[Lodash](https://lodash.com/docs#template) options can be passed in using the querystring or adding or a JSON object to your configuration.
```
Comment (okonet) [inline README.md]: Should Underscore be removed since you're aliasing `lodash`?

File: package.json [hunk 6]
```diff
@@ -23,7 +23,7 @@
   },
   "homepage": "https://github.com/okonet/ejs-loader",
   "dependencies": {
-    "lodash": "^3.6.0",
-    "loader-utils": "^0.2.7"
+    "loader-utils": "^1.1.0",
+    "lodash": "^4.17.4"
```
Comment (okonet) [inline package.json]: Should be a `peerDependency`

Comment (okonet) [reply package.json]: Let's make it also less strict by specifying it like `>= 4`

File: README.md [hunk 7]
```diff
@@ -6,43 +6,55 @@ If you are looking for the loader which uses [EJS templating engine](https://git
...
   }
 };
 ```
+
+Config example using a JSON object:
+``` js
+module.exports = {
+  module: {
+    rules: [
+      { test: /\.ejs$/, use: [{loader: 'ejs-loader', options: { variable: 'data' }}] },
```
Comment (okonet) [inline README.md]: Please reformat to make it more readable

Comment (okonet) [reply README.md]: Is this example even needed?

File: README.md [hunk 8]
```diff
@@ -6,43 +6,55 @@ If you are looking for the loader which uses [EJS templating engine](https://git
...
+[Underscore](http://underscorejs.org/#template)/[Lodash](https://lodash.com/docs#template) options can be passed in using the querystring or adding or a JSON object to your configuration.
 
 Config example using a querystring:
 ``` js
 module.exports = {
   module: {
-    loaders: [
-      { test: /\.ejs$/, loader: 'ejs-loader?variable=data' },
+    rules: [
+      { test: /\.ejs$/, use: [{loader: 'ejs-loader?variable=data'}] },
```
Comment (okonet) [inline README.md]: I don't think passing options via query string is supported in Webpack 3

Comment (okonet) [reply README.md]: Let's remove this example completely

File: README.md (hunk 4 above)
Comment (okonet) [reply README.md]: I'm not sure how this is gonna work with `lodash` so please update to the working example

File: README.md (hunk 3 above)
Comment (okonet) [reply README.md]: Please leave this part as it was

**Comment by turntayble81** [comment]
Is this project being maintained? It looks like no progress on this pr in nearly a year. I'm running webpack 4.15.1 and still getting this warning:

parseQuery() will be replaced with getOptions() in the next major version of loader-utils.

**Comment by okonet** [comment]
Not really. I’m happy to pass it over to you if you want

**Comment by AtofStryker** [comment]
@okonet would love to pick this repository up if you are not looking to maintain it anymore. Would definitely like to make some enhancement's to reflect improvements made in webpack 4.

**Comment by okonet** [comment]
Sure I can transfer it to you. I'll need to add you to the npm as well. What's your username there?

**Comment by AtofStryker** [comment]
The featured requested here should now be supported and implemented https://github.com/difelice/ejs-loader/pull/42 and is available in version `0.4.1`
